binding.cc

```diff
@@ -952,28 +995,46 @@ namespace zmq {
     if(zmq_socket_monitor(socket->socket_, addr, ZMQ_EVENT_ALL) != -1) {
       socket->monitor_socket_ = zmq_socket (context->context_, ZMQ_PAIR);
       zmq_connect (socket->monitor_socket_, addr);
-
+      socket->timer_interval_ = timer_interval;
+      socket->num_of_events_ = num_of_events;
       socket->monitor_handle_ = new uv_timer_t;
-
       socket->monitor_handle_->data = socket;
 
       uv_timer_init(uv_default_loop(), socket->monitor_handle_);
-      uv_timer_start(socket->monitor_handle_, reinterpret_cast<uv_timer_cb>(Socket::UV_MonitorCallback), timer_interval, timer_interval);
+      uv_timer_start(socket->monitor_handle_, reinterpret_cast<uv_timer_cb>(Socket::UV_MonitorCallback), timer_interval, 0);
     }
 
     NanReturnUndefined();
   }
 
+  void
+  Socket::Unmonitor() {
+    // Make sure we are monitoring
+    if (this->monitor_socket_ == NULL) {
+      return;
+    }
+
+    // Passing NULL as addr will tell zmq to stop monitor
+    zmq_socket_monitor(this->socket_, NULL, ZMQ_EVENT_ALL);
+
+    // Close the monitor socket and stop timer
+    if (zmq_close(this->monitor_socket_) < 0)
+      throw std::runtime_error(ErrorMessage());
```

Comment:
Hmmm. Good question. I actually don't know. Try and see some time... I guess it would lead to errors as you are not allowed to access V8 from arbitrary threads.

Anyway, I think the code is correct now.

On June 3, 2015 12:36:15 AM EEST, Bent Cardan <notifications@github.com> wrote:
>>      }
>>  
>>      NanReturnUndefined();
>>    }
>>  
>> +  void
>> +  Socket::Unmonitor() {
>> +    // Make sure we are monitoring
>> +    if (this->monitor_socket_ == NULL) {
>> +      return;
>> +    }
>> +
>> +    // Passing NULL as addr will tell zmq to stop monitor
>> +    zmq_socket_monitor(this->socket_, NULL, ZMQ_EVENT_ALL);
>> +
>> +    // Close the monitor socket and stop timer
>> +    if (zmq_close(this->monitor_socket_) < 0)
>> +      throw std::runtime_error(ErrorMessage());
>
>just meant that the method wasn't exposed to v8, in these cases can we
>still do `NanThrowError`?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/JustinTulloss/zeromq.node/pull/424/files#r31574017